core/charm/repository/charmhub_test.go

```diff
@@ -791,6 +806,27 @@ func (*selectNextBaseSuite) TestSelectNextBaseWithValidBases(c *gc.C) {
 	}})
 }
 
+func (*selectNextBaseSuite) TestSelectNextBaseWithValidBasesWithSeries(c *gc.C) {
+	repo := new(CharmHubRepository)
+	platform, err := repo.selectNextBases([]transport.Base{{
+		Architecture: "amd64",
+		Name:         "ubuntu",
+		Channel:      "focal",
```

Comment:
If you add another transport.Base with amd64/ubuntu/20.04 the result has duplicated entries. We've seen this in the actual data returning from charm hub. We need to guard against that.

Reply:
Fixed